### PR TITLE
BUG: DatetimeArray._from_sequence accepting bool dtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Identified when making the change this makes in Series.

### CODE REVIEW

File: pandas/tests/arrays/test_datetimes.py [hunk 1]
```diff
@@ -89,11 +89,26 @@ def test_non_array_raises(self):
...
 
-    def test_other_type_raises(self):
+    def test_bool_dtype_raises(self):
+        arr = np.array([1, 2, 3], dtype="bool")
+
         with pytest.raises(
             ValueError, match="The dtype of 'values' is incorrect.*bool"
         ):
-            DatetimeArray(np.array([1, 2, 3], dtype="bool"))
+            DatetimeArray(arr)
```
Comment (simonjayhawkins) [inline pandas/tests/arrays/test_datetimes.py]: why is this a ValueError when DatetimeArray._from_sequence(arr) is a TypeError? Is is also worth considering making error messages consistent?

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -1903,32 +1904,36 @@ def maybe_convert_dtype(data, copy):
     ------
     TypeError : PeriodDType data is passed
     """
-    if is_float_dtype(data):
+    if not hasattr(data, "dtype"):
```
Comment (simonjayhawkins) [inline pandas/core/arrays/datetimes.py]: is this part of the bugfix or additional cleanup?

Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py]: Cleanup for perf; my eye twitches whenever i see is_foo_dtype(values) that could be is_foo_dtype(values.dtype)

File: pandas/tests/arrays/test_datetimes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/test_datetimes.py]: originally i made it type error, then changed for consistenty with... something, but i dont remember what.  will try to figure that out and report bakc

**Comment by jreback** [comment]
looks fine. need a whatsnew as a bug fix for pd.to_datetime erroneously accepting bools

File: pandas/core/tools/datetimes.py [hunk 3]
```diff
@@ -361,7 +361,18 @@ def _convert_listlike_datetimes(
     # warn if passing timedelta64, raise for PeriodDtype
     # NB: this must come after unit transformation
     orig_arg = arg
-    arg, _ = maybe_convert_dtype(arg, copy=False)
+    try:
+        arg, _ = maybe_convert_dtype(arg, copy=False)
+    except TypeError:
+        if errors == "coerce":
```
Comment (jreback) [inline pandas/core/tools/datetimes.py]: actually do we have coverage for all of these paths? i seem to recall we using similar code elsewhere?

Comment (jbrockmendel) [reply pandas/core/tools/datetimes.py]: yes, we have tests in tests.tools.test_to_datetime that hit all of these cases

**Comment by jreback** [comment]
can you rebase your PRs again, now that azure if running. (likely not a big deal, but just to make sure)

File: pandas/core/tools/datetimes.py (hunk 3 above)
Comment (jreback) [reply pandas/core/tools/datetimes.py]: kk

also maybe able to de-duplicate some of this code

**Comment by jreback** [comment]
thanks

**Pull request merged**